src/storage/txn/scheduler.rs

```diff
@@ -139,80 +142,56 @@ impl TaskContext {
     }
 }
 
-/// Scheduler which schedules the execution of `storage::Command`s.
-pub struct Scheduler<E: Engine> {
-    engine: E,
-
+struct SchedulerInner<E: Engine> {
     // cid -> Task
-    pending_tasks: HashMap<u64, Task>,
+    pending_tasks: Vec<Mutex<HashMap<u64, Task>>>,
```

Comment:
I remember in my latest implementation, I have already removed pending_tasks, we can only use one hash map directly.